### DISCUSSION TITLE
Input file assumptions

### DISCUSSION THREAD

**Discussion by glennDittmann** [BrunoLevy/geogram#151] [Q&A] [answered]
Hi again,
I have a quick question on the `.pts` input files.

I would like to execute the example `compute_delaunay` with `dimension=2` and `dimension=3`, i.e. a 2d and a 3d delaunay triangulation.

I was passing an `.pts` file like this for `dimension=2`, 
```txt
v 0 1 0
v 1 0 0
v 2 2 0
...
 ```
assuming the algorithm would then interpret the third column as the weights of the points (`0`) and therefore compute the 2d delaunay triangulation over the first two  columns.

 
 However, when I wanted to pass something like this for `dimension=3`,
```txt
v 0 1 1 0
v 1 1 0 0
v 2 2 2 0
...
 ```
I get a `wrong number of fields` error.

How do I then pass weights for 3d points ? And is my assumption for the 2d case correct?

**Comment by BrunoLevy** [comment, marked as answer]
The example program `compute_delaunay` does not compute weighted triangulations, see the documentation [here](https://github.com/BrunoLevy/geogram/wiki/Delaunay3D) for an example of weighted triangulation.
Regarding file formats:
- if you use the full geogram library, then all file formats supported by geogram will work.
- if you use the PSM (pluggable software module), then it is ASCII, with d coordinates per line (d = 2 or d = 3)

**Reply by glennDittmann** [reply]
Thanks !
